File: eulerground-python/4.py
#find the largest prime factor of a composite number
import sys
if len(sys.argv) == 1:
    n = 600951475143L
elif len(sys.argv) == 2:
    n = eval(sys.argv[1])
else:
    print "Usage : python 4.py <number>"
    exit(0)

if n % 2 == 0:
    n = n / 2
    lastFactor = 2
    while n % 2 == 0:
        n = n / 2
else:
    lastFactor = 1L

factor = 3L
while n > 1:
    if n % factor == 0:
        n = n / factor
        lastFactor = factor
        while n % factor == 0:
            n = n / factor
    factor = factor + 2L
print lastFactor

File: eulerground-c++/4.cpp
//Largest prime factor of a number
#include <iostream>
#include <cmath>
#include <cstdio>
#include <cstdlib>
using namespace std;

int main(int argc,char ** argv)
{

    long long number;
    long long factor,lastFactor;
    if (argc < 2 )
        number = 600951475143;
    else if(argc == 2)
        number = atoi(argv[1]);
    else {
        cout << "Usage : ./a.out <number> "<<endl;
        exit(0);
        }
    cout << "Number : "<<number<<endl;
    if ( number%2 == 0)
        {number = number/2;
        while(number %2 == 0)
            number = number / 2;}
    else 
        lastFactor = 1;
    factor = 3;
    while (number > 1)
    {
        if(number % factor == 0)
            {number = number / factor;
             lastFactor = factor;
             while (number % factor == 0)
                number = number / factor;
            }
        factor = factor + 2;
    }
    cout << lastFactor;
    return 0;
}
    

    

